elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt

```diff
@@ -13,27 +15,26 @@ import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.isActive
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.sync.Mutex
-import kotlinx.coroutines.sync.withLock
 import money.vivid.elmslie.core.ElmScope
 import money.vivid.elmslie.core.config.ElmslieConfig
 
 @Suppress("TooGenericExceptionCaught")
+@OptIn(ExperimentalCoroutinesApi::class)
 class ElmStore<Event : Any, State : Any, Effect : Any, Command : Any>(
     initialState: State,
     private val reducer: StateReducer<Event, State, Effect, Command>,
     private val actor: Actor<Command, out Event>,
     storeListeners: Set<StoreListener<Event, State, Effect, Command>>? = null,
     override val startEvent: Event? = null,
     private val key: String =
-        (reducer::class.qualifiedName?: reducer::class.simpleName).orEmpty().replace(
+        (reducer::class.qualifiedName ?: reducer::class.simpleName).orEmpty().replace(
             "Reducer",
             "Store",
         ),
 ) : Store<Event, Effect, State> {
 
     private val logger = ElmslieConfig.logger
-    private val eventMutex = Mutex()
+    private val eventDispatcher = ElmslieConfig.ioDispatchers.limitedParallelism(parallelism = 1)
```

Comment:
to guarantee a sequence of events, it is better to use coroutines channels

Reply:
=) not really